server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/VariableNode.java

```diff
@@ -102,8 +102,8 @@ public SyntaxError getError(MessageTemplate messageTemplate, ErrorSeverity sever
   public void addUsage(VariableUsageNode usageNode) {
     if (!usages.contains(usageNode.getLocality().toLocation())) {
       usages.add(usageNode.getLocality().toLocation());
-      usageNode.addDefinition(this);
     }
+    usageNode.addDefinition(this);
```

Comment:
I am still investigating this. But I believe its a valid change, if the usage locality is in a copybook which is referred from multiple places.